### DISCUSSION TITLE
Bref courses ...

### DISCUSSION THREAD

**Discussion by karlkras** [brefphp/bref#1140] [General]
We're considering jumping into one of the bref courses in hopes of gaining a faster track to productive use.
Possibly the mid tier option.
Can someone give me an idea of how much time, given ~4 hours per day to dedicate to the training, a senior php developer (well versed in Laravel and lumen frameworks) but with moderate AWS experience can expect to work through this?
Are there "live" instructors who can provide support and direction if/when a participant  is confronted with a blocking implementation and/or design issue?

thanks for any feedback.

**Comment by deleugpn** [comment]
Hi @karlkras 

You may want to join our slack community and ask a few questions there. From time to time there are community members available to answer questions (see https://bref.sh/docs/community.html).

We also provide Business Support, you can fill out this form here to get in touch: https://bref.sh/#enterprise

As a general guidance, my personal opinion is that AWS can be a bit dauthing and complex, but Bref tries to simplify a lot of it by removing some options and definiting simpler paths to take. If your goal is to host your application on Lambda with limited AWS knowledge, 2~3 days should be more than enough to get started. However, as day to day activities start to emerge, debugging on AWS start to become pressing and understanding a few key concepts may come in handy. A deep, professional AWS knowledge might take between 1 to 6 months to build up excellence, depending on your definition of moderate AWS experience.

The course https://serverless-visually-explained.com/ provided by the creator of Bref can greatly help understand the serverless parts of AWS which can help prepare your team to understand how one can envision serverless.

One of the most common issues faced by software engineers when learning about Lambda is around NAT Gateway, Connectivity Timeout and RDS connection. 7777 (https://port7777.com) can reduce a lot of time when trying to connect to your RDS from your local machine by provisioning every resource necessary to establish a secure connection between your machine and your RDS. A recent change made by Napoli (https://github.com/brefphp/bref/pull/1133) also attempts to address the lack of visibility when connectivity issues arise.

If you don't want to buy port7777, I have a blog post here explaining how to set it up for yourself: https://blog.deleu.dev/running-a-bastion-host-on-fargate/
I have also written a comprehensive guide on AWS VPC for Software Engineers here: https://blog.deleu.dev/aws-vpc-for-software-engineers/
There is also a great curated list of PHP Serverless here: https://github.com/aws-samples/php-examples-for-aws-lambda/blob/master/serverless-php-resources.md



As with any new tech, start small and gradually build up complexity. Let me know if I managed to address any of your concerns and if there is anything else I can answer. Welcome to Bref and Serverless, I hope you enjoy it as much as I do!

**Reply by karlkras** [reply]
Thanks, but my question was specific to the bref courses and how comprehensive they are.
For example, would these challenges you outline be addressed with strategies on how to address?
We're not noobs to aws so we're not coming in totally ignorant, just looking to flatten the bref learning curve where possible.

**Reply by mnapoli** [reply]
Hi @karlkras. To complete on Marco's great answer above:

>  Are there "live" instructors who can provide support and direction if/when a participant is confronted with a blocking implementation and/or design issue?

No, the course doesn't include that.

>  We're not noobs to aws so we're not coming in totally ignorant, just looking to flatten the bref learning curve where possible.

To be honest, there's a lot to learn if you want to go to production, and the last thing I want to do is guarantee that if you put the course in the hands of anyone they'll master AWS and go to production in X days.

**Reply by karlkras** [reply]
Thanks.
